sympy/solvers/solveset.py

```diff
@@ -58,6 +58,11 @@
 from collections import defaultdict
 
 
+class NonLinearError(ValueError):
+    """Raised by linear_eq_to_matrix if the equations aren't linear"""
+    pass
```

Comment:
I think that the error should be `NonlinearError` (i.e. with a lower case `l` for linear) because nonlinear is a single word. Also "are not linear" or "are nonlinear".